default/AI/ShipDesignAI.py

```diff
@@ -0,0 +1,1602 @@
+"""
+This module deals with the autonomous shipdesign from the AI. The design process is class-based.
+The basic functionality is defined in the class AIShipDesign. The more specialised classes mostly 
+implement the rating function and some additional information for the optimizing algorithms to improve performance. 
+
+Example usage of this module:
+import ShipDesignAI
+myDesign = ShipDesignAI.AIShipDesign_Military()
+myDesign.additional_specifications.enemy_mine_dmg = 10
+best_military_designs = myDesign.optimize_design()  # best designs per planet: (rating,planetID,design_id,cost) tuples
+
+
+Available ship classes:
+- AIShipDesign_Military: basic military ship
```

Comment:
Regarding naming in general-- we are trying to gradually make the code as compliant as possible with [PEP8 naming conventions] (https://www.python.org/dev/peps/pep-0008/#package-and-module-names), and I see no point in introducing new deviations like CapWords_WithUnderscores.  As for the particulars here, I agree with Morlic that it seems like it would be a pointless source of potential confusion to have a distinct class named identically to one we are already using implicitly or explicitly via the fo module, distinguished only by the 'fo.' prefix and/or capitalization.  I would like to suggest "ShipDesigner" since I think that captures a little better the idea of what this class does. The subclasses could be things like "ColonyShipDesigner", "WarShipDesigner", etc.